traffic_ops/traffic_ops_golang/trafficvault/backends/postgres/postgres.go

```diff
@@ -100,24 +101,149 @@ func (p *Postgres) commitTransaction(tx *sqlx.Tx, ctx context.Context, cancelFun
 	cancelFunc()
 }
 
+// GetDeliveryServiceSSLKeys retrieves the SSL keys of the given version for
+// the delivery service identified by the given xmlID. If version is empty,
+// the implementation should return the latest version.
 func (p *Postgres) GetDeliveryServiceSSLKeys(xmlID string, version string, tx *sql.Tx, ctx context.Context) (tc.DeliveryServiceSSLKeysV15, bool, error) {
-	return tc.DeliveryServiceSSLKeysV15{}, false, notImplementedErr
+	tvTx, dbCtx, cancelFunc, err := p.beginTransaction(ctx)
+	if err != nil {
+		return tc.DeliveryServiceSSLKeysV15{}, false, err
+	}
+	defer p.commitTransaction(tvTx, dbCtx, cancelFunc)
+	var jsonKeys string
+	query := "SELECT data FROM sslkey WHERE deliveryservice=$1"
+	if version != "" {
+		query += " AND version=$2"
+		err = tvTx.QueryRow(query, xmlID, version).Scan(&jsonKeys)
+	} else {
+		err = tvTx.QueryRow(query, xmlID).Scan(&jsonKeys)
+	}
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return tc.DeliveryServiceSSLKeysV15{}, false, nil
+		}
+		e := checkErrWithContext("Traffic Vault PostgreSQL: executing SELECT SSL Keys query", err, ctx.Err())
+		return tc.DeliveryServiceSSLKeysV15{}, false, e
+	}
+	sslKey := tc.DeliveryServiceSSLKeysV15{}
+	err = json.Unmarshal([]byte(jsonKeys), &sslKey)
+	if err != nil {
+		return tc.DeliveryServiceSSLKeysV15{}, false, errors.New("unmarshalling ssl keys: " + err.Error())
+	}
+	return sslKey, true, nil
 }
 
+// PutDeliveryServiceSSLKeys stores the given SSL keys for a delivery service.
 func (p *Postgres) PutDeliveryServiceSSLKeys(key tc.DeliveryServiceSSLKeys, tx *sql.Tx, ctx context.Context) error {
-	return notImplementedErr
+	tvTx, dbCtx, cancelFunc, err := p.beginTransaction(ctx)
+	if err != nil {
+		return err
+	}
+	defer p.commitTransaction(tvTx, dbCtx, cancelFunc)
+
+	keyJSON, err := json.Marshal(&key)
```

Comment:
we should check to make sure it has a version provided and if it doesnt then read the db and increase from the latest

Reply:
I think callers of this method are still free to leave `version` empty, in which case we would write it to the DB w/ `version = "latest"`.

Reply:
So the `version` cannot be empty in this request, the validators make sure of that before passing the request body into this method. But I've made the change so that the methos actually insert two rows, one with the specified version, and one with `latest`, after deleting the `latest` and the key with the given version (if provided)